lua/cfc_random_spawn/module/sv_player_spawning.lua

```diff
@@ -7,25 +8,79 @@ if not mapHasCustomSpawns then return end
 
 CFCRandomSpawn.spawnPointRankings = CFCRandomSpawn.spawnPointRankings or {}
 
-local function getMeasurablePlayers()
+local function getMeasurablePlayers( respawner )
     local measurablePlayers = {}
     for _, ply in pairs( player.GetHumans() ) do
-        if ( ply:Alive() ) then
+        if ply:Alive() and respawner ~= ply then
             table.insert( measurablePlayers, ply )
         end
     end
 
     return measurablePlayers
 end
 
+
+-- Only use spawnpoints that are nearest to the most "popular" point
+-- This lets everyone spawn near the most active part of the server
+
+-- if count is hit and distance is not hit, more spawnpoints will be chosen & vice versa
+
+local nearSpawnpointsMinCount = 8 -- minimum count of spawnpoints that will be availiable
+local nearSpawnpointsMinDistance = 2500 -- minimum size of area that spawnpoints will be chosen from
+local nearSpawnpointsMinDistanceSqr = nearSpawnpointsMinDistance ^ 2
+
+local function sortSpawnsByDistance( comparePos, spawns )
+    table.sort( spawns, function( a, b ) -- sort by distance
+        local aDist = a.spawnPos:DistToSqr( comparePos )
+        local bDist = b.spawnPos:DistToSqr( comparePos )
+        return aDist < bDist
+    end )
+    return spawns
+end
+
+local function getNearestSpawns( nearPos, spawns )
+    local distSortedSpawns = sortSpawnsByDistance( nearPos, spawns )
+    local bestSpawn = distSortedSpawns[1] -- get best spawn so the distance comparison isnt worthless
+    local nearestSpawns = {}
+
+    for _, spawn in pairs( distSortedSpawns ) do
+        local distToFirstSqr = bestSpawn.spawnPos:DistToSqr( spawn.spawnPos ) -- will never run on every spawnpoint
+        local overCount = _ >= nearSpawnpointsMinCount
+        local overDistance = distToFirstSqr > nearSpawnpointsMinDistanceSqr
+
+        if overCount and overDistance then 
+            break
+        else
+            nearestSpawns[_] = spawn
```

Comment:
This will always return a single entry table, as far as i know, why would you need a single entry table?